### PR TITLE
Cleaned up new examples

### PR DESCRIPTION
**Pull request by artv3**
New examples for RAJA documentation. 
Example 1. Batched matrix multiply with different data layouts. 
Example 2. five-cell stencil with RAJA offsets.

### CODE REVIEW

File: examples/ex-batched-matrix-multiply.cpp [hunk 1]
```diff
@@ -52,239 +52,434 @@
...
 #endif
 
-//Dimensions of matrices
-const Index_type NCOLS = 3;
-const Index_type NROWS = 3;
-const Index_type NMAT  = 12000000;
+// Dimensions of matrices
+const int NCOLS = 3;
+const int NROWS = 3;
+//const int NMAT = 12000000;
```
Comment (davidbeckingsale) [inline examples/ex-batched-matrix-multiply.cpp]: Why is this line here?

File: examples/ex-batched-matrix-multiply.cpp [hunk 2]
```diff
@@ -27,264 +27,469 @@
...
+ *           A^{1}_{01}, A^{0}_{02}, A^{1}_{02},
+ *           A^{0}_{10}, A^{1}_{10}, A^{0}_{11},
+ *           A^{1}_{11}, A^{0}_{12}, A^{1}_{12},
+ *           A^{0}_{20}, A^{1}_{20}, A^{0}_{21},
+ *           A^{1}_{21}, A^{0}_{22}, A^{1}_{22}];
  *
- * We expect that layout 1 will perform better on the CPU
- * and layout 2 would perform better on the GPU.
+ * Since layout 1 has the entries for a matrix
+ * close in memory it simplifies vector operations.
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: More correctly, it 'allows' vector operations. Do we see SIMD actually generated by compilers?

File: examples/ex-batched-matrix-multiply.cpp [hunk 3]
```diff
@@ -27,264 +27,469 @@
...
+  double *Bl2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+  double *Cl2 = memoryManager::allocate<double>(NCOLS * NROWS * NMAT);
+
+  //
+  // Permuted layout - equivalent to indexing via
+  // A[e + NELEM*(r + NROWS*c)]
+  auto layout2 =
+      RAJA::make_permuted_layout({{NMAT, NROWS, NCOLS}},
+                                 RAJA::as_array<RAJA::Perm<1, 2, 0> >::get());
+  RAJA::View<double, RAJA::Layout<3, Index_type, 0> > Al2view(Al2, layout2);
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: It may be easier to read if you change these to 'Aview2', etc. What is the 'l' for?

File: examples/ex-batched-matrix-multiply.cpp [hunk 4]
```diff
@@ -27,264 +27,469 @@
...
-        Cview(i,1,0) = Aview(i,1,0)*Bview(i,0,0) + Aview(i,1,1)*Bview(i,1,0) + Aview(i,1,2)*Bview(i,2,0);
-        Cview(i,1,1) = Aview(i,1,0)*Bview(i,0,1) + Aview(i,1,1)*Bview(i,1,1) + Aview(i,1,2)*Bview(i,2,1);
-        Cview(i,1,2) = Aview(i,1,0)*Bview(i,0,2) + Aview(i,1,1)*Bview(i,1,2) + Aview(i,1,2)*Bview(i,2,2);
-        
-        Cview(i,2,0) = Aview(i,2,0)*Bview(i,0,0) + Aview(i,2,1)*Bview(i,1,0) + Aview(i,2,2)*Bview(i,2,0);
-        Cview(i,2,1) = Aview(i,2,0)*Bview(i,0,1) + Aview(i,2,1)*Bview(i,1,1) + Aview(i,2,2)*Bview(i,2,1);
-        Cview(i,2,2) = Aview(i,2,0)*Bview(i,0,2) + Aview(i,2,1)*Bview(i,1,2) + Aview(i,2,2)*Bview(i,2,2);
-
-      });
+    RAJA::forall<CPUPol1>(RAJA::RangeSegment(0, NMAT), [=](Index_type i) {
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: I think it's clearer if you just use the exec policy directly for the template parameter; i.e., no 'using' directive.

File: examples/ex-batched-matrix-multiply.cpp [hunk 5]
```diff
@@ -27,264 +27,469 @@
...
-        Cl2view(i,1,0) = Al2view(i,1,0)*Bl2view(i,0,0) + Al2view(i,1,1)*Bl2view(i,1,0) + Al2view(i,1,2)*Bl2view(i,2,0);
-        Cl2view(i,1,1) = Al2view(i,1,0)*Bl2view(i,0,1) + Al2view(i,1,1)*Bl2view(i,1,1) + Al2view(i,1,2)*Bl2view(i,2,1);
-        Cl2view(i,1,2) = Al2view(i,1,0)*Bl2view(i,0,2) + Al2view(i,1,1)*Bl2view(i,1,2) + Al2view(i,1,2)*Bl2view(i,2,2);
-        
-        Cl2view(i,2,0) = Al2view(i,2,0)*Bl2view(i,0,0) + Al2view(i,2,1)*Bl2view(i,1,0) + Al2view(i,2,2)*Bl2view(i,2,0);
-        Cl2view(i,2,1) = Al2view(i,2,0)*Bl2view(i,0,1) + Al2view(i,2,1)*Bl2view(i,1,1) + Al2view(i,2,2)*Bl2view(i,2,1);
-        Cl2view(i,2,2) = Al2view(i,2,0)*Bl2view(i,0,2) + Al2view(i,2,1)*Bl2view(i,1,2) + Al2view(i,2,2)*Bl2view(i,2,2);
-        
-      });
+    RAJA::forall<CPUPol1>(RAJA::RangeSegment(0, NMAT), [=](Index_type i) {
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same exec policy comment as above.

File: examples/ex-batched-matrix-multiply.cpp [hunk 6]
```diff
@@ -27,264 +27,469 @@
...
+  //-------------------------------------------
+  // Matrix multiply with layout 1 on the CPU with OpenMP threads
+  //
+  using CPUPol2 = RAJA::omp_parallel_for_exec;
+
+  myMin = std::numeric_limits<double>::max();
+  for (int i = 0; i < NITER; ++i) {
+
+    timer.start();
+    RAJA::forall<CPUPol2>(RAJA::RangeSegment(0, NMAT), [=](Index_type i) {
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same exec policy comment as above.

File: examples/ex-batched-matrix-multiply.cpp [hunk 7]
```diff
@@ -27,264 +27,469 @@
...
+  //-------------------------------------------
+
+  //-------------------------------------------
+  // Matrix multiply with layout 2 on the CPU with OpenMP parallel policy
+  //
+  myMin = std::numeric_limits<double>::max();
+  for (int i = 0; i < NITER; ++i) {
+
+    timer.start();
+    RAJA::forall<CPUPol2>(RAJA::RangeSegment(0, NMAT), [=](Index_type i) {
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same exec policy comment as above.

File: examples/ex-batched-matrix-multiply.cpp [hunk 8]
```diff
@@ -27,264 +27,469 @@
...
-        
-        Cview(i,1,0) = Aview(i,1,0)*Bview(i,0,0) + Aview(i,1,1)*Bview(i,1,0) + Aview(i,1,2)*Bview(i,2,0);
-        Cview(i,1,1) = Aview(i,1,0)*Bview(i,0,1) + Aview(i,1,1)*Bview(i,1,1) + Aview(i,1,2)*Bview(i,2,1);
-        Cview(i,1,2) = Aview(i,1,0)*Bview(i,0,2) + Aview(i,1,1)*Bview(i,1,2) + Aview(i,1,2)*Bview(i,2,2);
-        
-        Cview(i,2,0) = Aview(i,2,0)*Bview(i,0,0) + Aview(i,2,1)*Bview(i,1,0) + Aview(i,2,2)*Bview(i,2,0);
-        Cview(i,2,1) = Aview(i,2,0)*Bview(i,0,1) + Aview(i,2,1)*Bview(i,1,1) + Aview(i,2,2)*Bview(i,2,1);
-        Cview(i,2,2) = Aview(i,2,0)*Bview(i,0,2) + Aview(i,2,1)*Bview(i,1,2) + Aview(i,2,2)*Bview(i,2,2);
-      });
+    RAJA::forall<GPUPol>(RAJA::RangeSegment(0, NMAT),
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same exec policy comment as above.

File: examples/ex-batched-matrix-multiply.cpp [hunk 9]
```diff
@@ -27,264 +27,469 @@
...
-        Cl2view(i,1,0) = Al2view(i,1,0)*Bl2view(i,0,0) + Al2view(i,1,1)*Bl2view(i,1,0) + Al2view(i,1,2)*Bl2view(i,2,0);
-        Cl2view(i,1,1) = Al2view(i,1,0)*Bl2view(i,0,1) + Al2view(i,1,1)*Bl2view(i,1,1) + Al2view(i,1,2)*Bl2view(i,2,1);
-        Cl2view(i,1,2) = Al2view(i,1,0)*Bl2view(i,0,2) + Al2view(i,1,1)*Bl2view(i,1,2) + Al2view(i,1,2)*Bl2view(i,2,2);
-        
-        Cl2view(i,2,0) = Al2view(i,2,0)*Bl2view(i,0,0) + Al2view(i,2,1)*Bl2view(i,1,0) + Al2view(i,2,2)*Bl2view(i,2,0);
-        Cl2view(i,2,1) = Al2view(i,2,0)*Bl2view(i,0,1) + Al2view(i,2,1)*Bl2view(i,1,1) + Al2view(i,2,2)*Bl2view(i,2,1);
-        Cl2view(i,2,2) = Al2view(i,2,0)*Bl2view(i,0,2) + Al2view(i,2,1)*Bl2view(i,1,2) + Al2view(i,2,2)*Bl2view(i,2,2);
-      
-      });
+    RAJA::forall<GPUPol>(
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same exec policy comment as above.

File: examples/ex-offset.cpp [hunk 10]
```diff
@@ -13,70 +13,74 @@
...
- * Input: 
+ *  Example applies a ``five-box stencil" to
+ *  interior cells of a lattice. A five-box stencil
+ *  accumulates values of a box and its four neighbors.
+ *  The resulting values are used to populate the
+ *  interior cells of a second lattice.
+ *  We assume N x N interior nodes and a padded edge
+ *  of zeros for a lattice of size (N + 2) x (N + 2).
+ *
+ *  In the case of N = 2, the first lattice is generated
```
Comment (rhornung67) [inline examples/ex-offset.cpp]: Please recall the suggestion I made via email about using N = 3, here in the explanation as well as the example.

File: examples/ex-offset.cpp [hunk 11]
```diff
@@ -87,217 +91,228 @@
...
+  //
+  RAJA::OffsetLayout<DIM> layout = RAJA::make_offset_layout<DIM>({{-1, -1}}, {{N, N}});    
+  RAJA::View<int, RAJA::OffsetLayout<DIM>> lattice0view(lattice0, layout);
+  RAJA::View<int, RAJA::OffsetLayout<DIM>> lattice1view(lattice1, layout);
+
+  //----------------------------------------------------------------------------//
+  std::cout << "\n Running sequential five-box-stencil (RAJA-Kernel - "
+               "sequential)...\n";
+  using NESTED_EXEC_POL = RAJA::
+      KernelPolicy<RAJA::statement::
```
Comment (rhornung67) [inline examples/ex-offset.cpp]: Please format the nested policies (here and others below) using the indentation pattern and put each closing '>' on its own line as in the matrix multiplication examples in the ex4... file.  It makes the policies much easier to read and understand.

**Comment by artv3** [comment]
Thanks @rhornung67 , I was wondering how we should address the OMP first touch performance issue? Should we add initialize additional memory for OMP? Or just initialize the data using threads if OMP is enabled?

**Comment by rhornung67** [comment]
I think the simplest thing to do is to initialize the data using the same RAJA openmp loop traversal that you use for the calculation.

File: examples/ex-batched-matrix-multiply.cpp [hunk 12]
```diff
@@ -159,11 +159,11 @@ int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
 //
 // Initialize data
 //
-#if defined(RAJA_ENABLE_OPENMP)
+//#if defined(RAJA_ENABLE_OPENMP)
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: All OpenMP usage must be guarded with the RAJA_ENABLE_OPENMP macro.

File: examples/ex-batched-matrix-multiply.cpp [hunk 13]
```diff
@@ -27,299 +27,515 @@
...
+// Initialize data
 //
-  for(Index_type e=0; e<NMAT; ++e){
-    for(Index_type row=0; row<NROWS; ++row){
-      for(Index_type col=0; col<NCOLS; ++col){
-        Aview(e,row,col) = rand() % 50 + 1; 
-        Bview(e,row,col) = rand() % 50 + 1; 
-        Al2view(e,row,col) = Aview(e,row,col);
-        Bl2view(e,row,col) = Bview(e,row,col);
+//#if defined(RAJA_ENABLE_OPENMP)
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: This guard must be used

File: examples/ex-batched-matrix-multiply.cpp [hunk 14]
```diff
@@ -27,299 +27,515 @@
...
     timer.reset();
   }
-  std::cout<<"Matrix Multiplication with layout 1 on CPU run time : "<<myMin<<" seconds"<<std::endl;
-//-------------------------------------------
-  
-//-------------------------------------------
-//Matrix multiply with layout 2 on the CPU
-//
-  using CPUPol2 = RAJA::loop_exec;
+  std::cout << "Performing batched matrix multiplication with layout 1 using "
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Please add a blank line and separator between examples here like you do for the others.

Never mind. I see what your doing now. You should print the message about which example is run BEFORE it is run in case there is an issue.

File: examples/ex-batched-matrix-multiply.cpp [hunk 15]
```diff
@@ -27,299 +27,515 @@
...
     timer.reset();
   }
-  std::cout<<"Matrix Multiplication with layout 2 on CPU run time : "<<myMin<<" seconds"<<std::endl;
-  //---------------------------------------------
-  
-//
-//Compare output
-//
-  compareOutput(Cview, Cl2view, NMAT);
+  std::cout << "Performing batched matrix multiplication with layout 2 using "
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Please put the print message about which example is run before the example is run (same comment as above.

File: examples/ex-batched-matrix-multiply.cpp [hunk 16]
```diff
@@ -27,299 +27,515 @@
...
+                       + Aview(e, 2, 1) * Bview(e, 1, 2)
+                       + Aview(e, 2, 2) * Bview(e, 2, 2);
+    });
+    timer.stop();
+
+    RAJA::Timer::ElapsedType tMin = timer.elapsed();
+    if (tMin < minRun) minRun = tMin;
+    timer.reset();
+  }
+  std::cout << "Performing batched matrix multiplication with layout 1 using "
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same comment about moving print message.

File: examples/ex-batched-matrix-multiply.cpp [hunk 17]
```diff
@@ -27,299 +27,515 @@
...
+                        + Aview2(e, 2, 2) * Bview2(e, 2, 2);
+
+    });
+    timer.stop();
+
+    RAJA::Timer::ElapsedType tMin = timer.elapsed();
+    if (tMin < minRun) minRun = tMin;
+    timer.reset();
+  }
+  std::cout << "Performing batched matrix multiplication with layout 2 using "
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Move print message to before example is run.

File: examples/ex-batched-matrix-multiply.cpp [hunk 18]
```diff
@@ -27,299 +27,515 @@
...
-  std::cout<<"Matrix Multiplication with layout 1 on GPU run time : "<<myMin<<" seconds"<<std::endl;
-//-------------------------------------------
-  
-//-------------------------------------------
-//Matrix multiply with layout 2 on the GPU
-//
-  myMin = std::numeric_limits<double>::max();  
-  for(Index_type i=0; i<NITER; ++i){
-    
+  std::cout << "Matrix Multiplication with layout 1 on GPU with "
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Same comment about message location.

File: examples/ex-batched-matrix-multiply.cpp [hunk 19]
```diff
@@ -27,299 +27,515 @@
...
-  RAJA::Timer::ElapsedType tMin = timer.elapsed();
-  if(tMin < myMin) myMin = tMin;
-  timer.reset();
+
+    RAJA::Timer::ElapsedType tMin = timer.elapsed();
+    if (tMin < minRun) minRun = tMin;
+    timer.reset();
   }
-  std::cout<<"Matrix Multiplication with layout 2 on GPU run time : "<<myMin<<" seconds"<<std::endl;
+  std::cout << "Matrix Multiplication with layout 2 on GPU with "
```
Comment (rhornung67) [inline examples/ex-batched-matrix-multiply.cpp]: Move message to before example.

File: examples/ex-offset.cpp [hunk 20]
```diff
@@ -13,70 +13,91 @@
...
+ *  output_lattice(row, col)
+ *         = input_lattice(row, col)
+ *         + input_lattice(row - 1, col) + input_lattice(row + 1, col)
+ *         + input_lattice(row, col - 1) + input_lattice(row, col + 1)
+ *
+ *  We assume a lattice has N x N interior nodes 
+ *  and a padded edge of zeros for a lattice
+ *  of size (N + 2) x (N + 2).  
+ *
+ *  In the case of N = 3, the input lattice generated
```
Comment (rhornung67) [inline examples/ex-offset.cpp]: 👍
